server/model/build.go

```diff
@@ -28,6 +28,7 @@ type Build struct {
 	Error        string       `json:"error"                   xorm:"build_error"`
 	Enqueued     int64        `json:"enqueued_at"             xorm:"build_enqueued"`
 	Created      int64        `json:"created_at"              xorm:"build_created"`
+	Updated      int64        `json:"updated_at"              xorm:"updated NOT NULL DEFAULT 0 'updated'"`
```

Comment:
Could you explain why we need an additional `updated_at` from my understanding of the issue we would need to set `finished_at` after a pipeline was marked as failed.

Reply:
It's a kind a hotfix ... update will get updated by xorm (build in feature) on each insert/update

make finished_at set in each case would require a whole more code ... and we are in freeze
-> the issue is, that the gRPC Done do set this but only if not multipipeline aka procs list only contain one parent proc
-> else it cant set finished as it has no info of other pipelines can set finish or not (since we dont have a dependency graph based on procs)

Reply:
Hmm, that's a bummer.

To be able to remove the `updated_at` field easier again if needed after we properly fixed `finished_at` could you split this PR from the refactoring of the moved helper functions.

Reply:
and if we like to get rid of it, just delete the line ... + some migration to drop it

Reply:
so reverting via squash-commit wont be a good idea so why split this pull?
touched code is in the scope of this topic